pkg/applicationmanager/controllers/applicationregistration.go

```diff
@@ -71,6 +71,39 @@ func createAppReg(regCRD map[string][]stork_api.ApplicationResource) error {
 		if _, err := stork.Instance().CreateApplicationRegistration(appReg); err != nil && !errors.IsAlreadyExists(err) {
 			logrus.Errorf("unable to register app %v, err: %v", appReg, err)
 			return err
+		} else if errors.IsAlreadyExists(err) {
+			// Get the existing app reg
+			existingAppReg, err := stork.Instance().GetApplicationRegistration(appReg.Name)
+			if err != nil {
+				logrus.Errorf("unable to get an existing application registration")
+			} else {
+				needsRegistration := false
+				appRegName := appReg.Name
+				for _, newResource := range appReg.Resources {
+					found := false
+					for _, existingResource := range existingAppReg.Resources {
+						if newResource.Group == existingResource.Group &&
+							newResource.Kind == existingResource.Kind {
```

Comment:
Should we check for equality of versions as well?

Reply:
I think we should. Updating it to check the version too. Also updating the appReg name to include the version.